verifiable-db/src/revelation/revelation_unproven_offset.rs

```diff
@@ -0,0 +1,1295 @@
+                        .chain(index_node_info[i].node_max.to_targets())
+                        .chain(once(index_ids[0]))
+                        .chain(row_proof.index_value_target().to_targets())
+                        .chain(row_tree_root.to_targets())
+                        .collect_vec();
+                    b.hash_n_to_hash_no_pad::<H>(inputs)
+                };
+                let index_path_wires = MerklePathGadget::build(b, index_node_hash, index_ids[0]);
+                // if the current row is valid, check that the root is the same of the original tree, completing
+                // membership proof for the current row; otherwise, we don't care
+                let root = b.select_hash(is_matching_row, &index_path_wires.root, &tree_hash);
+                b.connect_hashes(tree_hash, root);
+
+                row_paths.push(row_path_wires.inputs);
+                index_paths.push(index_path_wires.inputs);
+                // check that the primary index value for the current row is within the query
+                // bounds (only if the row is valid)
+                let index_value = row_proof.index_value_target();
+                let greater_than_min = b.is_less_or_equal_than_u256(&min_query, &index_value);
+                let smaller_than_max = b.is_less_or_equal_than_u256(&index_value, &max_query);
+                let in_range = b.and(greater_than_min, smaller_than_max);
+                let in_range = b.and(is_matching_row, in_range);
+                b.connect(in_range.target, is_matching_row.target);
+
+                // enforce DISTINCT only for actual results: we enforce the i-th actual result is strictly smaller
+                // than the (i+1)-th actual result
+                max_result = if let Some(res) = &max_result {
+                    let current_result: [UInt256Target; S] =
+                        get_result(i).to_vec().try_into().unwrap();
+                    let is_smaller = b.is_less_than_or_equal_to_u256_arr(res, &current_result).0;
+                    // flag specifying whether we must enforce DISTINCT for the current result or not
+                    let must_be_enforced = b.and(is_matching_row, distinct);
+                    let is_smaller = b.and(must_be_enforced, is_smaller);
+                    b.connect(is_smaller.target, must_be_enforced.target);
+                    Some(current_result)
+                } else {
+                    Some(get_result(i).to_vec().try_into().unwrap())
+                };
+
+                // Expose results for this row.
+                // First, we compute the digest of the results corresponding to this row, as computed in the universal
+                // query circuit, to check that the results correspond to the one computed by that circuit
+                let cells_tree_hash =
+                    build_cells_tree(b, &get_result(i)[2..], &ids[2..], &is_item_included[2..]);
```

Comment:
But the results could be stuff like `balance * 10` for example, it wouldn't match the cell tree as stored in the tree, would it ? A bit lost on why are we re-creating that cell tree hash here since we pass in the results and not the raw values stored in our db.

Reply:
Because the universal circuit, which is employed to generate the proof for a single row, is still assuming that a results tree will be built for these type of queries. So it accumulates all the results related to this row in a cells tree, and the digest is computed from this cells tree. So here we are recomputing the cells tree in order to correctly recompute the digest. This allows to avoid introducing another variant of the universal circuit just for these type of queries, and instead to re-use the variant we will employ once we will have the results tree integrated

Reply:
Ahh makes sense thanks for the explanation. Could you put that quickly in a comment above ? Thanks.

Reply:
Done in commit https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/365/commits/c89fc0fe19d8241f14f3bcce0c8c21fd6fae34cb